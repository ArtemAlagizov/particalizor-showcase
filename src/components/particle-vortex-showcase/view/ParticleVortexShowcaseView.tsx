import React from "react";
import { IParticleVortexShowcaseView } from "./IParticleVortexShowcaseView";
import { ParticleVortex } from "particalizor-3000";
import { Styled } from "../styled";
import "antd/dist/antd.css";
import SliderWithLabel from "../../../ui-components/slider-with-label/container/SliderWithLabel";
import InputWithLabel from "../../../ui-components/input-with-label/container/InputWithLabel";

const ParticleVortexShowcaseView: React.FC<IParticleVortexShowcaseView> = ({
  onParticleNumberChange,
  onTraceWidthChange,
  onLifeTimeChange,
  onVortexNumberChange,
  onImageWidthChange,
  onImageHeightChange,
  onBackgroundColorChange,
  particleNumberValue,
  particleTraceWidthValue,
  particleLifeTimeValue,
  imageWidth,
  imageHeight,
  backgroundColor,
  vortexNumberValue
}) => {
  return (
    <Styled.ShowcaseComponentContainer>
      <Styled.PropertiesContainer>
        <Styled.PropertiesSet>
          <SliderWithLabel
            propertyName={"particleNumber"}
            selectedValue={particleNumberValue}
            onValueChange={onParticleNumberChange}
            label={"particle number"}
            range={{ min: 10, max: 1990 }}
            step={20}
          />
          <SliderWithLabel
            propertyName={"particleTraceWidthValue"}
            selectedValue={particleTraceWidthValue}
            onValueChange={onTraceWidthChange}
            label={"particle trace width"}
            range={{ min: 1, max: 800 }}
          />
          <SliderWithLabel
            propertyName={"particleLifeTimeValue"}
            selectedValue={particleLifeTimeValue}
            onValueChange={onLifeTimeChange}
            label={"particle life time"}
            range={{ min: 100, max: 7000 }}
            step={100}
          />
          <SliderWithLabel
            propertyName={"selectedVortexNumberValue"}
            selectedValue={vortexNumberValue}
            onValueChange={onVortexNumberChange}
            label={"vortex number"}
            range={{ min: 1, max: 20 }}
          />
        </Styled.PropertiesSet>
        <Styled.PropertiesSet>
          <SliderWithLabel
            propertyName={"imageWidth"}
            selectedValue={imageWidth}
            onValueChange={onImageWidthChange}
            label={"image width"}
            range={{ min: 200, max: 1200 }}
            step={100}
          />
          <SliderWithLabel
            propertyName={"imageHeight"}
            selectedValue={imageHeight}
            onValueChange={onImageHeightChange}
            label={"image height"}
            range={{ min: 200, max: 1200 }}
            step={100}
          />
          <InputWithLabel
            propertyName={"backgroundColor"}
            defaultValue={"#33344c"}
            label={"background color"}
            onValueChange={onBackgroundColorChange}
            selectedValue={backgroundColor}
          />
        </Styled.PropertiesSet>
      </Styled.PropertiesContainer>
      <Styled.Images>
        <Styled.ImageContainer>
          <ParticleVortex
            imageWidth={imageWidth}
            imageHeight={imageHeight}
            vortexNumber={vortexNumberValue}
            particleTraceWidth={particleTraceWidthValue}
            particleNumber={particleNumberValue}
            particleLifeTime={particleLifeTimeValue}
            backgroundColor={backgroundColor}
          />
        </Styled.ImageContainer>
      </Styled.Images>
    </Styled.ShowcaseComponentContainer>
  );
};

export default ParticleVortexShowcaseView;
