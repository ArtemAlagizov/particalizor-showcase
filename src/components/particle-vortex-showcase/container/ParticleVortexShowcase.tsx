import React, { useState } from "react";
import ParticleVortexShowcaseView from "../view/ParticleVortexShowcaseView";
import { SliderValue } from "antd/es/slider";

const ParticleVortexShowcase: React.FC = () => {
  const [
    selectedParticleNumberValue,
    setSelectedParticleNumberValue
  ] = useState<number>(30);
  const [
    selectedParticleTraceWidthValue,
    setSelectedParticleTraceWidthValue
  ] = useState<number>(600);
  const [
    selectedParticleLifeTimeValue,
    setSelectedParticleLifeTimeValue
  ] = useState<number>(1100);
  const [selectedVortexNumberValue, setSelectedVortexNumberValue] = useState<
    number
  >(3);
  const [selectedImageHeight, setSelectedImageHeight] = useState<number>(500);
  const [selectedImageWidth, setSelectedImageWidth] = useState<number>(900);
  const [selectedBackgroundColor, setSelectedBackgroundColor] = useState<
    string
  >("#33344c");

  const onChange = (value: SliderValue) =>
    setSelectedParticleNumberValue(value as number);

  const onTraceWidthChange = (value: SliderValue) =>
    setSelectedParticleTraceWidthValue(value as number);

  const onLifeTimeChange = (value: SliderValue) =>
    setSelectedParticleLifeTimeValue(value as number);

  const onVortexNumberChange = (value: SliderValue) =>
    setSelectedVortexNumberValue(value as number);

  const onImageWidthChange = (value: SliderValue) =>
    setSelectedImageWidth(value as number);

  const onImageHeightChange = (value: SliderValue) =>
    setSelectedImageHeight(value as number);

  const onBackgroundColorChange = (value: string) =>
    setSelectedBackgroundColor(value as string);

  return (
    <ParticleVortexShowcaseView
      onLifeTimeChange={onLifeTimeChange}
      onParticleNumberChange={onChange}
      onTraceWidthChange={onTraceWidthChange}
      particleLifeTimeValue={selectedParticleLifeTimeValue}
      particleNumberValue={selectedParticleNumberValue}
      particleTraceWidthValue={selectedParticleTraceWidthValue}
      vortexNumberValue={selectedVortexNumberValue}
      onVortexNumberChange={onVortexNumberChange}
      onImageWidthChange={onImageWidthChange}
      onImageHeightChange={onImageHeightChange}
      onBackgroundColorChange={onBackgroundColorChange}
      backgroundColor={selectedBackgroundColor}
      imageWidth={selectedImageWidth}
      imageHeight={selectedImageHeight}
    />
  );
};

export default ParticleVortexShowcase;
