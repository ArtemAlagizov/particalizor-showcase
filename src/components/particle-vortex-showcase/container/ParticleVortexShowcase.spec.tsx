import React from "react";
import { mount, shallow } from "enzyme";
import ParticleVortexShowcase from "./ParticleVortexShowcase";
import ParticleVortexShowcaseView from "../view/ParticleVortexShowcaseView";

describe("Given a ParticleVortexShowcase", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(<ParticleVortexShowcase />);

      expect(component).toMatchSnapshot();
    });
    it("should render ParticleVortexShowcaseView", () => {
      const component = mount(<ParticleVortexShowcase />);

      expect(component.find(ParticleVortexShowcaseView).length).toBe(1);
    });
  });
});
