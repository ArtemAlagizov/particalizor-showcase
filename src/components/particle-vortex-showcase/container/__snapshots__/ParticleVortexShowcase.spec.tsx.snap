// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ParticleVortexShowcase when rendering should match snapshot 1`] = `
<ParticleVortexShowcaseView
  backgroundColor="#33344c"
  imageHeight={500}
  imageWidth={900}
  onBackgroundColorChange={[Function]}
  onImageHeightChange={[Function]}
  onImageWidthChange={[Function]}
  onLifeTimeChange={[Function]}
  onParticleNumberChange={[Function]}
  onTraceWidthChange={[Function]}
  onVortexNumberChange={[Function]}
  particleLifeTimeValue={1100}
  particleNumberValue={30}
  particleTraceWidthValue={600}
  vortexNumberValue={3}
/>
`;
